src/ripple/protocol/impl/SecretKey.cpp

```diff
@@ -88,16 +88,29 @@ signDigest (PublicKey const& pk, SecretKey const& sk,
     if (publicKeyType(pk.slice()) != KeyType::secp256k1)
         LogicError("sign: secp256k1 required for digest signing");
 
-    int siglen = 72;
-    unsigned char sig[72];
-    auto const result = secp256k1_ecdsa_sign(
-        secp256k1Context(),
-            digest.data(), sig, &siglen,
-                sk.data(), secp256k1_nonce_function_rfc6979,
-                    nullptr);
-    if (result != 1)
+    BOOST_ASSERT(sk.size() == 32);
```

Comment:
Its an assert because `SecretKey` is sized at compile time to 32 bytes. No need to incur a runtime penalty for something which is impossible.